### PR TITLE
Ap/poisson

### PR DESCRIPTION
**Pull request by APaganini**
Added a Poisson example, this will then become a test

### CODE REVIEW

File: examples/poisson/L2tracking_PDEconstraint.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import firedrake as fd
+from fireshape import PdeConstraint
+
+__all__ = ["PoissonSolver"]
```
Comment (florianwechsung) [inline examples/poisson/L2tracking_PDEconstraint.py]: Why do we need this?

**Review by florianwechsung — COMMENTED** [review]
Why is there an `__init__.py` in this example but not the other ones?

**Comment by APaganini** [comment]
This *pull-request* requires serious editing. DO NOT APPROVE IT NOW.

**Comment by APaganini** [comment]
@florianwechsung This pull request is ready to be merged, once `ap/docu-levelset` is merged.
This pull request:
- automates shape differentiation
- modifies the levelset example to use automated shape differentiation
- creates the example L2tracking and its documentation

**Comment by APaganini** [comment]
@florianwechsung  `pytest test_L2tracking.py`works locally, but fails on travis. Any idea?

**Comment by florianwechsung** [comment]
Had to update the firedrake docker image, that's now done and things seem to work.

**Pull request merged**